streams/readable-streams/async-iterator.any.js

```diff
@@ -338,3 +335,10 @@ promise_test(async () => {
   assert_equals(readResult.value, 3, 'should read remaining chunk');
   await reader.closed;
 }, 'Acquiring a reader and reading the remaining chunks after partially async-iterating a stream with preventCancel = true');
+
+promise_test(async t => {
+  const rs = new ReadableStream();
+  const it = rs.getIterator();
+  await it.return();
+  return promise_rejects(t, new TypeError(), it.next(), 'next() should reject');
+}, 'calling next() after return() should reject');
```

Comment:
Should we also test what happens if you do these two tests (return-return and return-next) without awaiting the return? I think that is also determined, right?

Reply:
Instead I added a test that the lock is released synchronously by return(), which is roughly equivalent and slightly improves coverage.